docs/distribution/fallback-launcher-release-infrastructure.md

```diff
@@ -0,0 +1,143 @@
+---
+layout: developer-doc
+title: Fallback Launcher Release Infrastructure
+category: distribution
+tags: [distribution, launcher, fallback]
+order: 6
+---
+
+# Fallback Launcher Release Infrastructure
+
+This document describes the fallback infrastructure that can be enabled to keep
+launcher updates functioning even if the primary release provider stops working.
+
+<!-- MarkdownTOC levels="2,3" autolink="true" -->
+
+- [Fallback Mechanism in the Launcher](#fallback-mechanism-in-the-launcher)
+- [Fallback Infrastructure Specification](#fallback-infrastructure-specification)
+  - [Requirements Inherited from Current Launcher Upgrade Scheme](#requirements-inherited-from-current-launcher-upgrade-scheme)
+- [Current Fallback Infrastructure Implementation](#current-fallback-infrastructure-implementation)
+  - [Updating the Release List](#updating-the-release-list)
+  - [Marking the Release As Broken](#marking-the-release-as-broken)
+
+<!-- /MarkdownTOC -->
+
+## Fallback Mechanism in the Launcher
+
+The launcher has a built-in fallback mechanism for its self-upgrades. If a
+request to the default release provider (currently GitHub Releases) fails, the
+fallback mechanism is queried to see if it is enabled and available. If it is
+enabled, the original request is re-tried with it. So in case the primary
+release provider becomes permanently disabled, the fallback mechanism can be
+enabled allowing launcher updates to function without forcing the users to do
+the update manually.
+
+Currently a fallback mechanism for engine releases is not provided, since if the
+engine release provider becomes obsolete, a new launcher version can be released
+that will know about the new engine provider.
```

Comment:
Probably worth mentioning that there aren't any plans to have a fallback for engine releases.